src/models/serialization.jl

```diff
@@ -195,8 +207,8 @@ function convert_type(
     return T{component_type}(values...)
 end
 
-function convert_type(::Type{T}, data::Any) where T <: Deterministic
-    error("This form of convert_type is not supported for Deterministic")
+function convert_type(::Type{T}, data::Any) where T <: Forecast
```

Comment:
Does this error imply that if a user uses a custom forecast object then it needs to also define the custom `convert_type`?

Reply:
No.  I defined the function ```convert_type(::Type{T}, data::Any) where T``` in src/utils/utils.jl.  It will convert any type except for a Forecast or Service.  If a caller wants to convert one of those then they have to call the version specific to those types, which take other parameters.

This is only necessary because the functions have different parameter lists.